Tools/ardupilotwaf/cxx_checks.py

```diff
@@ -157,26 +157,46 @@ def check_librt(cfg, env):
     return ret
 
 @conf
+def check_package(cfg, env, libname):
+    '''use pkg-config to look for an installed library that has a LIBNAME.pc file'''
+    cfg.check_cfg(package=libname, mandatory=False, global_define=True,
+                  args=['--libs', '--cflags'])
```

Comment:
Since those libraries are probably supposed to be used by all of our products, I think your current approach is fine  - I mean, adding the library variables to the global scope - although not quite ideal. What I meant is that it's safer to use `uselib_store=capsname` in order to correcly the the environment variables (e.g. `cfg.env['LIB_%s' % capsname]`). That should work even if Waf decides to changes the default uselib_store value in the future.

What I think might probably be better, but can be done in the future, is saying which program groups will use this library. That way we don't have to pass the local environments to the global scope.

The "correct" way to do things would be using the `use` kw argument on the task generators that use this library. But, again, I think we can be a little bit more relaxed on this matter and change this in the future.